src/Features/Core/Portable/EditAndContinue/EditAndContinueDiagnosticUpdateSource.cs

```diff
@@ -57,11 +57,15 @@ public ValueTask<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(Workspace w
         /// Clears all diagnostics reported thru this source.
         /// We do not track the particular reported diagnostics here since we can just clear all of them at once.
         /// </summary>
-        public void ClearDiagnostics()
+        public void ClearDiagnostics(bool isSessionEnding = false)
         {
             // If ClearDiagnostics is called and there weren't any diagnostics previously, then there is no point incrementing
             // our version number and potentially invalidating caches unnecessarily.
-            if (_previouslyHadDiagnostics)
+            // If the debug session is ending, however, we want to always increment otherwise we can get stuck. eg if the user
```

Comment:
>debug [](http://example.com/codeflow?start=22&length=5)

Can we always increment the version, to simplify this?

Reply:
We can, but it felt like maybe it was a bit too aggressive. We'd be essentially invalidating the cache that the LSP handler uses on every debugger step, even if there were no document changes.